astropy/nddata/nddata.py

```diff
@@ -234,12 +234,17 @@ def __init__(self, data, uncertainty=None, mask=None, wcs=None,
         self.uncertainty = uncertainty
 
     def __str__(self):
-        return str(self.data)
+        data = str(self.data)
+        unit = f" {self.unit}" if self.unit is not None else ''
+
+        return data + unit
 
     def __repr__(self):
         prefix = self.__class__.__name__ + '('
-        body = np.array2string(self.data, separator=', ', prefix=prefix)
-        return ''.join([prefix, body, ')'])
+        data = np.array2string(self.data, separator=', ', prefix=prefix)
+        unit = f", unit='{self.unit}'" if self.unit is not None else ''
+
+        return ''.join([prefix, data, unit, ')'])
```

Comment:
Using a f-string would be clearer, e.g. `f'{self.__class__.__name__}({data}{unit})`

Reply:
Agreed!

Reply:
I missed it, but the prefix is necessary fo indent levels and this makes the .join method the most succinct.

Reply:
Oh indeed, I  missed that too.